MetaMask/metamask-extension
# Fix for randomly resets of updated values in network edit form

## Explanation



Currently, in the network edit form, the field value is resetting randomly

This is a problem because the selectedNetwork prop is updated randomly which will cause calling the resetForm function and all field variables will reset.

In order to solve this problem, this pull request will add a new state **isEditing** which will not allow resetting of the form if the user changes something in the input field.

## More information



Fixes #13168

## Screenshots/Screencaps






### After


<img width="986" alt="image" src="https://user-images.githubusercontent.com/97883527/161912323-13fa00c3-c79c-476a-a196-7b7ce1e66972.png">


## Manual testing steps



Add a chain on https://chainlist.org/ like Optimistic Ethereum
Go to Settings > Networks > [the chain you just added]
Try editing the RPC to be https://localhost:8871/.
Wait ~10 seconds
Notice that your updated settings didn't change.

Cancel, Delete, Save, and switching between networks should also be tested.
